pandas/tests/frame/test_sorting.py

```diff
@@ -500,6 +526,71 @@ def test_sort_index_categorical_index(self):
         expected = df.iloc[[2, 3, 0, 1, 5, 4]]
         tm.assert_frame_equal(result, expected)
 
+    def test_sort_index_key(self):
+        df = DataFrame(np.arange(6, dtype="int64"), index=list("aaBBca"))
+
+        result = df.sort_index()
+        expected = df.iloc[[2, 3, 0, 1, 5, 4]]
+        tm.assert_frame_equal(result, expected)
+
+        result = df.sort_index(key=str.lower)
+        expected = df.iloc[[0, 1, 5, 2, 3, 4]]
+        tm.assert_frame_equal(result, expected)
+
+        result = df.sort_index(key=str.lower, ascending=False)
+        expected = df.iloc[[4, 2, 3, 0, 1, 5]]
+        tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("dtype", ["int8", "int64", "float64"])
+    def test_sort_index_key_int(self, dtype):
+        df = DataFrame(np.arange(6, dtype=dtype), index=np.arange(6, dtype=dtype))
+
+        result = df.sort_index()
+        tm.assert_frame_equal(result, df)
+
+        result = df.sort_index(key=lambda x: -x)
+        expected = df.sort_index(ascending=False)
+        tm.assert_frame_equal(result, expected)
+
+        result = df.sort_index(key=lambda x: 2 * x)
+        tm.assert_frame_equal(result, df)
+
+    def test_sort_value_key(self):
```

Comment:
nitpick: missing an `s` on values. Rename to `test_sort_values_key` (here and elsewhere).

Reply:
fixed in `frame/test_sorting.py` and `series/test_sorting.py`.